tango_simlib/tango_sim_generator.py

```diff
@@ -154,17 +154,21 @@ class object that the device server will inherit from
 
         """
         attr = attribute(label=attr_meta['label'], dtype=attr_meta['data_type'],
-                         enum_labels=sorted(attr_meta['enum_labels'])
-                         if 'enum_labels' in attr_meta.keys() else '',
+                         enum_labels=attr_meta['enum_labels']
+                         if 'enum_labels' in attr_meta else '',
                          doc=attr_meta['description'],
                          dformat=attr_meta['data_format'],
                          max_dim_x=attr_meta['max_dim_x'],
                          max_dim_y=attr_meta['max_dim_y'],
                          access=getattr(AttrWriteType, attr_meta['writable']))
         attr.__name__ = attr_name
         # Attribute read method
-        def read_meth(cls):
-            return getattr(cls, '_{}'.format(attr_name))
+        def read_meth(cls, attr):
```

Comment:
- [ ] Question: When read_meth() and attr.write are called, is `cls` actually a class, or really an instance of the Tango device server? If it is really an instance the parameter name `cls` is a bit confusing / misleading.